add fedora iot support to setup_selinux

bcambl opened the pull request:
#### Proposed Changes ####

Fix installation fail on Fedora 39 IoT (rpm-ostree).
```
[bcambl@k3s ~]$ curl -sfL https://get.k3s.io | sh -
[sudo] password for bcambl: 
[INFO]  Finding release for channel stable
[INFO]  Using v1.28.5+k3s1 as release
[INFO]  Downloading hash https://github.com/k3s-io/k3s/releases/download/v1.28.5+k3s1/sha256sum-arm64.txt
[INFO]  Skipping binary downloaded, installed k3s matches hash
[INFO]  Finding available k3s-selinux versions
sudo: yum: command not found
sudo: yum: command not found
```

On Fedora CoreOS the value of _VARIANT_ID_ within `/etc/os-release` is _coreos_, but on Fedora IoT the value is _iot_ resulting in the _setup_selinux_ function to attempt to use _yum_ instead of _rpm-ostree_.


#### Types of Changes ####

BugFix

#### Verification ####

This PR adds an additional condition to accommodate Fedora IoT along side the existing CoreOS rpm-ostree conditions. The proposed change in this PR checks for both _fedora_ and _iot_ values to ensure this additional condition is strictly for Fedora IoT.

Fedora 39 IoT `/etc/os-release` example:
```
[bcambl@k3s ~]$ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="39.20240111.0 (IoT Edition)"
ID=fedora
VERSION_ID=39
VERSION_CODENAME=""
PLATFORM_ID="platform:f39"
PRETTY_NAME="Fedora Linux 39.20240111.0 (IoT Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:39"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f39/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=39
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=39
SUPPORT_END=2024-11-12
VARIANT="IoT Edition"
VARIANT_ID=iot
OSTREE_VERSION='39.20240111.0'
```

#### Testing ####

The only references to _rpm-ostree_ are found within _install.sh_ and there are no existing tests for rpm-ostree based installs. I have uninstalled k3s and re-installed with this patch successfully:

```
# ./install.sh 
[INFO]  Finding release for channel stable
[INFO]  Using v1.28.5+k3s1 as release
[INFO]  Downloading hash https://github.com/k3s-io/k3s/releases/download/v1.28.5+k3s1/sha256sum-arm64.txt
[INFO]  Skipping binary downloaded, installed k3s matches hash
[INFO]  Finding available k3s-selinux versions
Checking out tree 09416ce... done
Enabled rpm-md repositories: updates fedora-cisco-openh264 rancher-k3s-common-stable fedora
Importing rpm-md... done
rpm-md repo 'updates' (cached); generated: 2024-01-19T02:33:12Z solvables: 16707
rpm-md repo 'fedora-cisco-openh264' (cached); generated: 2023-03-14T10:57:01Z solvables: 4
rpm-md repo 'rancher-k3s-common-stable' (cached); generated: 2023-06-02T16:56:56Z solvables: 2
rpm-md repo 'fedora' (cached); generated: 2023-11-01T00:12:29Z solvables: 61705
Resolving dependencies... done
Will download: 1 package (21.2 kB)
Downloading from 'rancher-k3s-common-stable'... done
Importing packages... done
Checking out packages... done
Running pre scripts... done
Running post scripts... done
Running posttrans scripts... done
Writing rpmdb... done
Writing OSTree commit... done
Staging deployment... done
Added:
  k3s-selinux-1.4-1.coreos.noarch
Changes queued for next boot. Run "systemctl reboot" to start a reboot
[WARN]  Please reboot your machine to activate the changes and avoid data loss.
[INFO]  Creating /usr/local/bin/kubectl symlink to k3s
[INFO]  Creating /usr/local/bin/crictl symlink to k3s
[INFO]  Creating /usr/local/bin/ctr symlink to k3s
[INFO]  Creating killall script /usr/local/bin/k3s-killall.sh
[INFO]  Creating uninstall script /usr/local/bin/k3s-uninstall.sh
[INFO]  env: Creating environment file /etc/systemd/system/k3s.service.env
[INFO]  systemd: Creating service file /etc/systemd/system/k3s.service
[INFO]  systemd: Enabling k3s unit
Created symlink /etc/systemd/system/multi-user.target.wants/k3s.service → /etc/systemd/system/k3s.service.
```

#### Linked Issues ####

Similar but unrelated issue related to Fedora CoreOS: https://github.com/k3s-io/k3s/issues/7666

#### User-Facing Change ####

```release-note
Fix install for Fedora IoT
```

#### Further Comments ####

dereknola commented:
You need to update the https://github.com/k3s-io/k3s/blob/master/install.sh.sha256sum with the new checksum
```
sha256sum install.sh > install.sh.sha256sum
```

bcambl commented:
Added a couple more checks for fedora iot alongside coreos in the _policy_ section of `setup_selinux()`.  
I have uninstalled, performed an `rpm-ostree reset`, reinstalled and updated the test output in this PR.
